baybe/recommenders/base.py

```diff
@@ -193,16 +173,18 @@ def _recommend_hybrid(
         """
         raise NotImplementedError("Hybrid recommendation is not implemented.")
 
-    def _select_candidates_and_recommend(
+    def _recommend_with_discrete_parts(
         self,
-        subspace_discrete: SubspaceDiscrete,
+        searchspace: SearchSpace,
         batch_size: int,
     ) -> pd.DataFrame:
-        """Get candidates in a discrete search space and generate recommendations.
+        """Obtain recommendations in search spaces with discrete parts.
```

Comment:
"with a discrete part", search spaces do not have mutliple discrete parts.

Reply:
depends on whether you count parts of parts also as parts :D

changed